modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/validation/ParameterTypeModelValidator.java

```diff
@@ -59,7 +96,20 @@ public void visitObject(ObjectType objectType) {
                                                                                           parameter.getName(),
                                                                                           getType(objectType).getName(),
                                                                                           String.class.getName()))));
+
+          objectType.getOpenRestriction().get().accept(this);
+        } else {
+          ExtensionMetadataTypeUtils.getType(objectType)
```

Comment:
These kind of validations are not compatible with AST Mode. 
Use Type wrappers to retrieve all the fields and types.